spring-jdbc/src/main/java/org/springframework/jdbc/support/GeneratedKeyHolder.java

```diff
@@ -61,26 +62,34 @@ public GeneratedKeyHolder(List<Map<String, Object>> keyList) {
 	@Override
 	@Nullable
 	public Number getKey() throws InvalidDataAccessApiUsageException, DataRetrievalFailureException {
+		return getKeyAs(Number.class);
+	}
+
+	@Override
+	@Nullable
+	public <T> T getKeyAs(Class<T> keyClass) throws InvalidDataAccessApiUsageException, DataRetrievalFailureException {
 		if (this.keyList.isEmpty()) {
 			return null;
 		}
 		if (this.keyList.size() > 1 || this.keyList.get(0).size() > 1) {
 			throw new InvalidDataAccessApiUsageException(
 					"The getKey method should only be used when a single key is returned.  " +
-					"The current key entry contains multiple keys: " + this.keyList);
+							"The current key entry contains multiple keys: " + this.keyList);
```

Comment:
Please revert this change.

In general, try not to change existing code (including reformatting) if you did not actually change the behavior. Otherwise, such formatting changes add too much noise to the PR and make it harder for people to review the changes.